core/common/src/main/java/com/alipay/sofa/rpc/common/utils/StringUtils.java

```diff
@@ -457,4 +457,29 @@ public static String substringBefore(String str, String separator) {
         }
         return str.substring(0, pos);
     }
+
+    public static boolean isNumeric(String str) {
+        if (str == null) {
+            return false;
+        } else {
+            int length = str.length();
+
+            for (int i = 0; i < length; ++i) {
+                if (!Character.isDigit(str.charAt(i))) {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+    }
+
+    public static Integer parseInteger(String str) {
+        if (isNumeric(str)) {
```

Comment:
`isNumeric` will return true when the string is empty, you need to add the `!isEmpty` judgment here.